homework sprint 11

leskiyy opened the pull request:
Реализовано ТЗ проекта №11

leskiyy committed:
homework sprint 11

leskiyy committed:
Merge branch 'main' into add-friends-likes

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/ErrorHandler.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.Map;
+
+@Slf4j
+@ControllerAdvice("ru.yandex.practicum.filmorate.controller")
```
Это ограничивает обработку исключений только в контроллерах. Если исключения будут выброшены в сервисах, они не будут обработаны. Лучше удалить параметр пакета

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/controller/ErrorHandler.java:
```diff
@@ -0,0 +1,43 @@
+    public ResponseEntity<Map<String, String>> handleValidation(final ValidationException e) {
+        log.error(e.getMessage());
+        return ResponseEntity
+                .status(HttpStatus.FORBIDDEN)
```
Статус кода для ValidationException установлен как HttpStatus.FORBIDDEN, хотя логичнее использовать BAD_REQUEST

Sla-als commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,68 @@
+        return filmStorage.removeLike(id, userId);
+    }
+
+    public List<Film> getPopularFilms(@Positive int count) {
```
В getPopularFilms отсутствует проверка на отрицательное значение count. Лучше добавить

leskiyy replied:
протестил отрицательные значения count в постмане @Positive аннатация работает корректо. в каких случаях может не сработать?

Sla-als reviewed (commented):
Нужны будут небольшие доработки, но в целом достойная работа

leskiyy committed:
homework sprint 11
fix: удалить параметр пакета аннотации @ControllerAdvice класса ErrorHandler
fix: поменять статус код обработки ValidationException на BAD_REQUEST

leskiyy committed:
Merge remote-tracking branch 'origin/add-friends-likes' into add-friends-likes
